tests/Unit/Domain/Test_GridCoordinates.cpp

```diff
@@ -0,0 +1,294 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Mesh.hpp"
+#include "Domain/CoordinateMaps/AffineMap.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/Direction.hpp"
+#include "Domain/GridCoordinates.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+SPECTRE_TEST_CASE("Unit.Domain.GridCoordinates", "[Domain][Unit]") {
+  using affine_map_2d =
+      CoordinateMaps::ProductOf2Maps<CoordinateMaps::AffineMap,
+                                     CoordinateMaps::AffineMap>;
+  using affine_map_3d =
+      CoordinateMaps::ProductOf3Maps<CoordinateMaps::AffineMap,
+                                     CoordinateMaps::AffineMap,
+                                     CoordinateMaps::AffineMap>;
+
+  /// [grid_coordinates_example]
```

Comment:
Only show 3D example, no output needed but if you want to show it, just show a couple CHECKs